Add toggle item loop feature

joelbrostrom opened the pull request:
By popular demand I created a simple bool, `loopItems`,
when set to false will reduce the item count to items.count.

This results in a static length page viewer that still retains
all the other benefits of this library (animations, buttons, autoplay etc...).

I also added comments to this and the vertical scroll feature i rebased against.
I realized now that it still have a PR open, but I added some info about it in the
Readme and change-log.

I suggest we add the features together and bump the version code to 1.2.0
(major.minor.patch) insteadof 1.2.0 and then 1.3.0 imminently after.

localpcguy commented:
First, thanks for implementing this! It works great, except it seems like something in this commit breaks the index property reported in the `onPageChanged` callback. I'll look a little deeper and see if I can determine what it is, but wanted to let you know right away. I don't see the issue in the v1.1.0 version.

joelbrostrom commented:
What does your onPageChange function look like?
You need to pass an int as an argument to invoke the function.
Wont work: `onPageChanged: () => print("OnPageChanged")`
Will work: `onPageChanged: (_) => print("OnPageChanged")`
I hard typed the function to remove ambiguous intentions.

joelbrostrom commented:
I Changed 
`Function onPageChanged`
to
`Function(int index) onPageChanged`

Does this solve your error?

localpcguy commented:
That wasn't it, turns out it was because I was setting a `currentCard` state property and using that for `initialPage` that was causing the widget to re-render on card swipes. Not sure why it wasn't happening in the v1.1.0 branch, but I don't think it was an error in the widget code after all. Thanks for taking a look and for the suggestions and sorry I didn't catch that prior to posting.

serenader2014 commented:
Thank you for your contribution, again! Since I'm quite busy these days, so appreciate your work.

But I have a simple advice, maybe changing the param name to `enableInfiniteScroll` is more suitable for this situation?

joelbrostrom commented:
Good point!
Implemented suggested changes, rebased and pushed the code.

If I can relieve of some of the work, consider adding me as a collaborator or a team member.

joelbrostrom commented:
Is there any issues with the PR or haven't you had the time to review it?
Please let me know if there is something I can do.
(I actually need this feature in a product I'm working on and thought it would be fun to use this lib instead of just ripping code it).

serenader2014 commented:
Actually I'm planning to merge the #24 first, and @showgp is working on the documentation. Or I can merge this PR first, and merge #24 later.

serenader2014 commented:
I have added you as a collaborator, so now you have permission to merge PR.

joelbrostrom commented:
Ok, I agree to let @showgp merge first if he is still active.
I rebased on master so I have the vertical feature but not the enlargeCenterPage fix in my branch.
I also added documentation to the vertical scroll feature, but I think it's fair to let the author add his own notes if he wants to.

@showgp I suggest you either: 
1. Write the docs and then merge. 
2. Merge the fix without writing docs and I will use my docs. 
3. Let me rebase on your pull request and then merge the enlargeCenterPage and the enableInfinite Scroll via this PR.

Sounds good?

serenader2014 commented:
@joelbrostrom Since #24 is not active, I think merging this PR would be OK.

joelbrostrom commented:
Ok, I'm in the alps skiing until Monday and won't be able to do it until then.
If you find time feel free to do it, otherwise I'll look into it once Im back :)
